packages/sensors/ios/Classes/SensorsPlugin.m

```diff
@@ -64,6 +77,50 @@ - (FlutterError*)onCancelWithArguments:(id)arguments {
 
 @end
 
+@implementation FLTUserAccelStreamHandler
+
+- (FlutterError*)onListenWithArguments:(id)arguments eventSink:(FlutterEventSink)eventSink {
+  _initMotionManager();
+  [_motionManager
+      startDeviceMotionUpdatesToQueue:[[NSOperationQueue alloc] init]
+                          withHandler:^(CMDeviceMotion* data, NSError* error) {
+                            CMAcceleration acceleration = data.userAcceleration;
+                            // Multiply by gravity, and adjust sign values to align with Android.
+                            sendTriplet(-acceleration.x * GRAVITY, -acceleration.y * GRAVITY,
+                                        -acceleration.z * GRAVITY, eventSink);
+                          }];
+  return nil;
+}
+
+- (FlutterError*)onCancelWithArguments:(id)arguments {
+  [_motionManager stopDeviceMotionUpdates];
+  return nil;
+}
+
+@end
+
+@implementation FLTGravityStreamHandler
+
+- (FlutterError*)onListenWithArguments:(id)arguments eventSink:(FlutterEventSink)eventSink {
+  _initMotionManager();
+  [_motionManager
```

Comment:
I tried this on a device and saw no data coming from the gravity handler.

Both the gravityhandler and the useracceleration handler use the "deviceMotionUpdates" and it cannot have two listeners.

We need to somehow register which streams are listened to and handle both of them in the same listener.

Likewise in the cancel method, we should only call [_motionManager stopDeviceMotionUpdates] if we have no more listeners. Using "reference" counting or something similar.